### PR TITLE
Feat/tour events observable

### PR DESCRIPTION
**Pull request by kriswinbush**
## Description

Created tourEvent$ observable and step/url Map for router transitions
### What's included?

- tourEvent$ observable
- step/url Map for tour route transitions
- going back with navigation demo
- changed dismissButton to voidButton and set display: none
#### Test Steps

- [ ] `npm run serve`
- [ ] select guided tour / examples
- [ ] run going back with navigation example

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

##### Screenshots or link to StackBlitz/Plunker
tour-event

### CODE REVIEW

File: src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-go-back-navigation/guided-tour-demo-go-back-navigation.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div id="go-back-navigation-demo">
+  <button mat-raised-button color="accent" (click)="startTour()">Start tour</button>
+  <h3 id="step-1">Step 1</h3>
+  <button mat-raised-button id="advance-button" [routerLink]="['/']">Step 2, click to navigate to step 3</button>
+  <h3 id="step-3">Step 4</h3>
```
Comment (thomaslennan) [inline src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-go-back-navigation/guided-tour-demo-go-back-navigation.component.html:5]: Change id to reflect step number? ie. step-4?

Comment (thomaslennan) [reply src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-go-back-navigation/guided-tour-demo-go-back-navigation.component.html]: Changed and pushed.

**Comment by philgetz** [comment]
nice!

File: src/platform/guided-tour/guided-tour.service.ts [hunk 2]
```diff
@@ -34,7 +33,7 @@ export interface IGuidedTourStep extends ITourStep {
 @Injectable()
 export class CovalentGuidedTourService extends CovalentGuidedTour {
   private _toursMap: Map<string, IGuidedTour> = new Map<string, IGuidedTour>();
-
+  private _tourStepURLs: Map<string, string> = new Map();
```
Comment (aqupriyanka) [inline src/platform/guided-tour/guided-tour.service.ts:36]: good to add generics for the new object

Comment (kriswinbush) [reply src/platform/guided-tour/guided-tour.service.ts:36]: done

File: src/platform/guided-tour/guided-tour.service.ts [hunk 3]
```diff
@@ -48,6 +47,10 @@ export class CovalentGuidedTourService extends CovalentGuidedTour {
       });
   }
 
+  tourEvent$(str: string): Observable<any> {
```
[the remark this answers is not in the document]
Comment (kriswinbush) [reply src/platform/guided-tour/guided-tour.service.ts:50]: done

File: src/platform/guided-tour/guided-tour.service.ts [hunk 4]
```diff
@@ -60,10 +63,27 @@ export class CovalentGuidedTourService extends CovalentGuidedTour {
       // remove steps from tour since we need to preprocess them first
       this.newTour(Object.assign({}, guidedTour, { steps: undefined }));
       const tourInstance: Shepherd.Tour = this.shepherdTour.addSteps(
-        this._configureRoutesForSteps(
-          this._prepareTour(guidedTour.steps, guidedTour.finishButtonText, guidedTour.dismissButtonText),
-        ),
+        this._configureRoutesForSteps(this._prepareTour(guidedTour.steps, guidedTour.finishButtonText)),
       );
+      // init route transition if step URL is different then the current location.
+      this.tourEvent$('show').subscribe((tourEvent: any) => {
```
[the remark this answers is not in the document]
Comment (kriswinbush) [reply src/platform/guided-tour/guided-tour.service.ts:69]: done

**Comment by aqupriyanka** [comment]
Could you please update the API documentation too

**Pull request merged**